src/core/api/index.js

```diff
@@ -60,15 +61,18 @@ export function callApi({
   const apiURL = `${API_BASE}/${endpoint}/${queryString}`;
 
   return fetch(apiURL, options)
-    .then((response) => response.json().then(
-      (jsonResponse) => ({ response, jsonResponse }),
-      (error) => {
-        log.warn('Could not parse response as JSON:', error);
-        return response.text().then((textResponse) =>
-          ({ response, jsonResponse: { text: textResponse } })
+    .then((response) => {
+      return response.clone().json()
+        .then(
+          (jsonResponse) => ({ response, jsonResponse }),
+          (error) => {
```

Comment:
The same code won't work with catch. If I put a catch above this and recover, it will execute the `jsonResponse` then function. I don't think there's another way to do it.

These kind of patterns is why promise chains are terrible! If we move to async/await (https://github.com/mozilla/addons-frontend/issues/1473) everything will be way easier to read.